File: largestPrimeFactor/largestPrimeFactor.cpp
#include <iostream>
#include <math.h>
using namespace std;

void primeFactor(long int num){  
    while (num % 2 == 0){  
        cout << 2 <<endl;  
        num = num/2;
    }
    for (int i = 3; i <= sqrt(num); i = i + 2){  
        while (num % i == 0){  
            cout << i <<endl;  
            num = num/i;  
        }  
    }  
    if (num > 2){
        cout << num <<endl;  
    }
}  

int main(){
    long int num =600851475143;
    primeFactor(num);
    return 0;
}

File: js/3-largestPrimeFactor/largestPrimeFactor.js
function primeFactor(num){
    let primeArr = [];
    while(num % 2 === 0){
        primeArr.push(2);
        num = num/2;
    }
    for (let i = 3; i <= num; i+=2){
        while(num % i === 0){
            primeArr.push(i);
            num = num/i;
        }
    }
    return primeArr;
}

console.log(Math.max(...primeFactor(600851475143)));